lte/gateway/python/magma/pipelined/app/enforcement.py

```diff
@@ -432,22 +437,37 @@ def _deactivate_flow_for_rule(self, imsi, rule_id):
             self.logger.error('Could not find rule id %s', rule_id)
             return
         cookie, mask = (num, flows.OVS_COOKIE_MATCH_ALL)
-        match = MagmaMatch(imsi=encode_imsi(imsi))
+
+        ip_match_in = get_ue_ipv4_match_args(ip_addr, Direction.IN)
+        match = MagmaMatch(eth_type=ether_types.ETH_TYPE_IP,
+                           imsi=encode_imsi(imsi), **ip_match_in)
+        flows.delete_flow(self._datapath, self.tbl_num, match,
+                          cookie=cookie, cookie_mask=mask)
+        ip_match_out = get_ue_ipv4_match_args(ip_addr, Direction.OUT)
+        match = MagmaMatch(eth_type=ether_types.ETH_TYPE_IP,
+                           imsi=encode_imsi(imsi), **ip_match_out)
         flows.delete_flow(self._datapath, self.tbl_num, match,
                           cookie=cookie, cookie_mask=mask)
         self._redirect_manager.deactivate_flow_for_rule(self._datapath, imsi,
                                                         num)
         self._qos_mgr.remove_subscriber_qos(imsi, num)
 
-    def _deactivate_flows_for_subscriber(self, imsi):
+    def _deactivate_flows_for_subscriber(self, imsi, ip_addr):
         """ Deactivate all rules for a subscriber, ending any enforcement """
```

Comment:
with multi apn the same subscriber can have multiple ip address, so this comment is no longer valid.